.github/workflows/bump.yml

```diff
@@ -0,0 +1,89 @@
+name: bump
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        required: true
+        description: Version to bump to
+
+env:
+  BUMP_BRANCH_NAME: bump-${{ inputs.version }}-${{ github.run_id }}
+
+jobs:
+  # Ensure that the provided version is greater than the current version
+  validate:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Get current version
+        id: current-version
+        run: |
+          props_file=$(cat Directory.Build.props)
+          version=$(echo "$props_file" | grep -oPm1 "(?<=<Version>)[^<]+")
+          echo "version=$version" >> $GITHUB_OUTPUT
+
+      - name: Validate new version
+        run: |
+          if npx semver "${{ inputs.version }}" --range "<= ${{ steps.current-version.outputs.version }}"; then
+            echo "New version must be greater than the current version"
+            exit 1
+          fi
```

Comment:
This uses the `semver` tool to make sure that the new version is valid and greater than the current version